scrapy/core/downloader/handlers/http10.py

```diff
@@ -1,17 +1,23 @@
 """Download handlers for http and https schemes
 """
 from twisted.internet import reactor
-from scrapy.utils.misc import load_object, create_instance
+
+from scrapy.utils.misc import create_instance, load_object
 from scrapy.utils.python import to_unicode
 
 
-class HTTP10DownloadHandler(object):
+class HTTP10DownloadHandler:
     lazy = False
 
-    def __init__(self, settings):
-        self.HTTPClientFactory = load_object(settings['DOWNLOADER_HTTPCLIENTFACTORY'])
-        self.ClientContextFactory = load_object(settings['DOWNLOADER_CLIENTCONTEXTFACTORY'])
+    def __init__(self, settings, crawler=None):
+        self.HTTPClientFactory = load_object(crawler.settings['DOWNLOADER_HTTPCLIENTFACTORY'])
+        self.ClientContextFactory = load_object(crawler.settings['DOWNLOADER_CLIENTCONTEXTFACTORY'])
```

Comment:
I'd prefer to use settings here, because crawler.settings doesn't exist when crawler=None; if we ever enable mypy, it'll bark at it. 

Maybe this None default doesn't work at all already - though I hope it should work in some cases, maybe with some features missing.

Reply:
I had it like that, but lost it between stashes. Thanks for pointing it out :ok_hand: